node/resolvers/checkout/index.ts

```diff
@@ -54,8 +72,30 @@ export const fieldResolvers = {
 }
 
 export const queries: Record<string, Resolver> = {
-  orderForm: (root, args, {dataSources: {checkout}}) => {
-    return checkout.orderForm()
+  orderForm: async (root, args, ctx) => {
+    const {dataSources: {checkout, session}, request: { headers: { cookie } }} = ctx
+
+    const cookieParsed = parse(cookie)
+    const coOrderFormId = cookieParsed['checkout.vtex.com'] && cookieParsed['checkout.vtex.com'].split('=')[1]
+    const sessionData = await sessionQueries.getSession(root, args, ctx) as SessionFields
+    if (sessionData.orderFormId) {
+      if (!coOrderFormId) {
+        ctx.request.headers.cookie = `${cookie}; checkout.vtex.com=__ofid=${sessionData.orderFormId}`
+      } else if (sessionData.orderFormId !== coOrderFormId) {
+        await session.editSession('orderFormId', coOrderFormId)
+      }
+    }
+
+    const orderForm = await checkout.orderForm()
+
+    if (!sessionData.orderFormId) {
```

Comment:
This needs to be synced with the last comment. I think in some cases it would happen twice.